apprise/plugins/NotifyNtfy.py

```diff
@@ -0,0 +1,331 @@
+
+    # A URL that takes you to the setup/help of the specific protocol
+    setup_url = 'https://github.com/caronc/apprise/wiki/Notify_ntfy'
+
+    # Default host if none is defined
+    default_host = 'ntfy.sh'
+
+    # Message time to live (if remote client isn't around to receive it)
+    time_to_live = 2419200
+
+    # Define object templates
+    templates = (
+        '{schema}://{topic}',
+        '{schema}://{host}/{topic}',
+        '{schema}://{host}:{port}/{topic}',
+    )
+
+    # Define our template tokens
+    template_tokens = dict(NotifyBase.template_tokens, **{
+        'topic': {
+            'name': _('Topic'),
+            'type': 'string',
+            'required': True,
+        },
+        'host': {
+            'name': _('Hostname'),
+            'type': 'string',
+        },
+        'port': {
+            'name': _('Port'),
+            'type': 'int',
+            'min': 1,
+            'max': 65535,
+        },
+    })
+
+    # Define our template arguments
+    template_args = dict(NotifyBase.template_args, **{
+        'priority': {
+            'name': _('Priority'),
+            'type': 'choice:string',
+            'values': NTFY_PRIORITIES,
+            'default': NtfyPriority.DEFAULT,
+        },
+        'tags': {
+            'name': _('Tags'),
+            'type': 'string',
+        },
+    })
+
+    def __init__(self, topic, priority=None, tags=None, **kwargs):
+        """
+        Initialize Ntfy Object
+        """
+        super(NotifyNtfy, self).__init__(**kwargs)
+        if not topic:
+            msg = 'A topic name must be provided.'
+            self.logger.warning(msg)
+            raise TypeError(msg)
+
+        self.topic = topic
+
+        # The priority of the message
+        if priority not in NTFY_PRIORITIES:
+            self.priority = self.template_args['priority']['default']
+        else:
+            self.priority = priority
+
+        # Any optional tags to attach to the notification
+        self.__tags = parse_list(tags)
+
+        # prepare our fullpath
+        self.fullpath = kwargs.get('fullpath')
+        if not isinstance(self.fullpath, six.string_types):
+            self.fullpath = '/'
+
+    def send(self, body, title='', notify_type=NotifyType.INFO, **kwargs):
+        """
+        Perform Ntfy Notification
+        """
+
+        # error tracking (used for function return)
+        has_error = False
+
+        # Prepare our headers
+        headers = {}
+        priority = NtfyPriority.get_priority(self.priority)
+        if priority != NtfyPriority.DEFAULT:
+            headers['X-Priority'] = priority
+        if title:
+            headers['X-Title'] = title
+        if self.__tags:
+            headers['X-Tags'] = ",".join(self.__tags)
+
+        # Prepare our payload
+        payload = body
+
+        # Prepare our Ntfy URL
+        schema = 'https' if self.secure else 'http'
+
+        # Use default host if one is not defined
+        host = self.default_host
```

Comment:
I don't know where else to comment, but this does not work as a host, but it should: `ntfys://--joey`, or `ntfys://joey--`

Reply:
What spec are you looking at for that? That would be a host of `joey--` with no topic, which doesn't seem valid. In fact, as a shortcut, I currently have that going to the default `ntfy.sh` host with a topic of `joey--` instead.

Either way, as a hostname it would be invalid according to RFC 952.

Reply:
It's not a hostname. It's a ntfy topic name. 

ntfys://joey-- translates to https://ntfy.sh/joey-- which is a valid topic name. The spec is my regex of allowed topic names: https://github.com/binwiederhier/ntfy/blob/main/server/server.go#L58

I think you are right though that apprise is using a URI parser which doesn't recognize this as a valid host and doesn't allow it, since joey-- in the previous example would be parsed as a hostname.

Reply:
Hmm. I felt like it would be a nice shortcut, but it inadvertently restricts topic names to conform to hostname standards. I'll see if I can fix it, otherwise we may have to be explicit about host 😕